### ISSUE TITLE
WSL ignores environment path

### ISSUE THREAD

**Issue by ezxpro** [microsoft/WSL#8976]
Long story short, I installed anaconda and added it to path using putting `export PATH=$PATH:/path/to/anaconda` in `.bashrc`, didn't work. on ZSH, I added it to `.zshrc`, didn't work either. These always work on a live distro, but not inside wsl. I then tried adding it to `/etc/profile`, didn't work either. 

Then someone told me about this [ask ubuntu issue](https://askubuntu.com/questions/60218/how-to-add-a-directory-to-the-path?noredirect=1&lq=1), which basically says you should add paths to `/etc/environment`.

I did it and it worked, until I restarted WSL and it no longer worked. 

I have none of this issues when running a full linux distro on my pc, not sure what's happening, but I'm sure it's a WSL issue.

**Comment by ezxpro** [comment]
So, after reading issues https://github.com/microsoft/WSL/issues/8932 https://github.com/microsoft/WSL/issues/816 and https://github.com/microsoft/WSL/issues/1405

**Comment by tusharsnn** [comment]
Can you share your `$PATH` so that we can have a look. May be some other program is modifying the values.

I am assuming none of the windows path is getting added if its disabled.

**Comment by ezxpro** [comment]
@OneBlue I found it inconsiderate of you to close the issue without even checking what's going on. You basically just dismissed everything I said as if it didn't matter and you automatically knew what's going on in my machine.

As to the configuration files, you were right. I deleted Ubuntu's `.zshrc` and  `.bashrc , then I was able to export the PATH alone for either zsh or bash. However, it doesn't work with the other configurations, so I'm left with a very basic shell that doesn't have autocomplete, coloring and other useful stuff, and I lack the expertise to diagnose where the conflict is.

But I still have my point here. I normally don't use `.zshrc`  or `.bashrc` for exporting paths. I learned that it's better to set it in `/etc/profile` because that normally reflects on your shell independently of user account or shell used. 

I'd like to know your thoughts on why Ubuntu ignores this file on WSL.
You also said this is absolutely not a wsl issue. It might be, but there are a few things here that are bugging me and I'd like some support. You sounded a lot like you just don't want to deal with any of this and chose to single-handedly close the issue. But there are a few things jumping to my eyes that I think are worth noticing:

1 wsl **does** concatenate `/usr/lib/wsl/lib` to the contents of my `/etc/environment`. I didn't find documentation for this behavior, which implied that, contrary to what you say, wsl still messes with the path even with interops disabled
2. It completely ignores the contents of `/etc/profile`, I tested this on a live Ubuntu 20.04 system and this behavior doesn't happen.
3. I reinstalled WSL Ubuntu 20.04 a couple times, Ubuntu 18.04, and confirmed that this issue only happens in WSL after I upgrade the system using `apt-get upgrade `. After the fresh installation and before the upgrade, if I set my path to `/etc/environment` it reflects just fine on both zsh and bash.

In conclusion, this looks like a pretty nasty bug to deal with and I understand it. It exists though, and maybe something can be done about it, and I'm willing to help track it down if that's possible. 



So it is clear that WSL at least adds its own thing to the PATH in the background and I couldn't find documentation for this behavior, and I'm unable to understand `why` it keeps ignoring `/etc/profile`
